<?xml version="1.0" encoding="UTF-8"?>
<test-data
    xmlns="http://pmd.sourceforge.net/rule-tests"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
    <test-code>
        <description><![CDATA[
bad
     ]]></description>
        <expected-problems>1</expected-problems>
        <code><![CDATA[
public class EmptyTryBlock1 {
       public void foo() {
               try {
               } catch (Exception e) {
                       e.printStackTrace();
               }
       }
}
     ]]></code>
    </test-code>
    <test-code>
        <description><![CDATA[
bad
     ]]></description>
        <expected-problems>1</expected-problems>
        <code><![CDATA[
public class EmptyTryBlock2 {
       public void foo() {
               try {
               } finally {
                       int x = 5;
               }
       }
}
     ]]></code>
    </test-code>
    <test-code>
        <description><![CDATA[
ok
     ]]></description>
        <expected-problems>0</expected-problems>
        <code><![CDATA[
public class EmptyTryBlock3 {
       public void foo() {
               try {
                       int f =2;
               } finally {
                       int x = 5;
               }
       }
}
     ]]></code>
    </test-code>
    <test-code>
        <description>#432 false positive for empty try-with-resource</description>
        <expected-problems>0</expected-problems>
        <code><![CDATA[
class X {
	void method() {
		try (ClientResponse response = execute(() -> target.request(mediaTypes).delete(), DELETE, new ExpectedResponse(status, required))) {
			// false positive
		}
	}
}
     ]]></code>
    </test-code>
</test-data>
